platform/security/certificate/src/main/java/org/codice/ddf/certificate/CertificateManager.java

```diff
@@ -0,0 +1,267 @@
+     ||
+     ||  Inherits From:  None
+     ||
+     ||     Interfaces:  None
+     ||
+     |+-----------------------------------------------------------------------
+     ||
+     ||      Constants:  MILLIS_IN_DAY - Number of milliseconds in a day
+     ||                  MILLIS_IN_YEAR - Number of milliseconds in a year
+     ||
+     |+-----------------------------------------------------------------------
+     ||
+     ||  Class Methods:     X509v3CertificateBuilder    createDefaultX509SigningRequest(PublicKey)
+     ||                     X509v3CertificateBuilder    createCustomX509SigningRequest(String, String, int, PublicKey)
+     ||                     X509v3CertificateBuilder    createCustomX509SigningRequest(String, String, String, String, String, String, String, String, String, String, int, PublicKey)
+     ||                     X509Certificate             signX509CertificateSHA256RSA(X509v3CertificateBuilder, PrivateKey)
+     ||                     void                        registerBCSecurityProvider()
+     ||                     String                      getHostName()
+     ||
+     ||
+     ++-----------------------------------------------------------------------*/
+
+public class CertificateManager {
+    private static final long MILLIS_IN_DAY = 86400000;
+
+    private static final long MILLIS_IN_YEAR = 31536000000L;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CertificateManager.class);
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: createDefaultX509SigningRequest
+     * <p/>
+     * Purpose:  Creates a signing request with default issuer and subject information
+     * <p/>
+     * Pre-condition: None
+     * <p/>
+     * Post-condition: Issuer information, subject information, and the encoded public key have been added to the returned signing request. Valid for 3 years.
+     *
+     * @param pubKey the PublicKey to be associated with the certificate signing request
+     * @return X509v3CertificateBuilder with the issuer, subject, and encoded public key set.
+     * -------------------------------------------------------------------
+     */
+
+    public static X509v3CertificateBuilder createDefaultX509SigningRequest(PublicKey pubKey) {
+        X509v3CertificateBuilder certBuilder;
+        Long currentTimeMillis = System.currentTimeMillis();
+        X500Name theIssuer = new X500Name(
+                "C=US,ST=AZ,O=DDF,OU=Dev,CN=DDF Demo Root CA,E=ddfrootca@example.org");
+        X500Name theSubject = new X500Name(
+                "CN=" + getHostName() + ", OU=Conn, OU=Dev, OU=DevT, O=U.S., C=US");
+        certBuilder = new X509v3CertificateBuilder(theIssuer,
+                new BigInteger(currentTimeMillis.toString()),
+                (new Date(currentTimeMillis - MILLIS_IN_DAY)),
+                (new Date(currentTimeMillis + (MILLIS_IN_YEAR * 3))), theSubject,
```

Comment:
We should probably be using Joda Time http://www.joda.org/joda-time/apidocs/org/joda/time/DateTime.html for all of this stuff. Then you can get rid of the constants up at the top o the class as well.